[Bug]: _handle_executable_exists fails when only python3 is on PATH

## Description

`_handle_executable_exists` in `handlers.py` uses `shutil.which(target)` to check for executables. When the rule target is `"python"`, the check fails on systems where only `python3` is available (common on modern Linux, CI environments).

This causes the test `test_executable_exists_found` to fail:
```
AssertionError: assert 'fail' == 'pass'
```

## Steps to Reproduce

1. Run on a system where `python` is not symlinked but `python3` exists
2. Execute: `python3 -m pytest tests/test_handler_registry.py::test_executable_exists_found`
3. Test fails because `shutil.which("python")` returns `None`

## Expected Behavior

The handler should detect Python via common aliases (`python3`) or the current interpreter (`sys.executable`) as fallback.

## Suggested Fix

In `_handle_executable_exists`, add fallback logic:
```python
found = shutil.which(target) is not None
if not found and target == "python":
    found = shutil.which("python3") is not None
if not found and target == "python":
    found = os.path.basename(sys.executable).startswith("python")
```

## Environment

- Python 3.10+ (system without `python` symlink)
- Affects test suite and real diagnostics

--- a/src/azure_functions_doctor/handlers.py
+++ b/src/azure_functions_doctor/handlers.py
@@ -576,7 +576,11 @@ class HandlerRegistry:
         target = condition.get("target")
         if not target:
             return _create_result("fail", "Missing 'target' for executable_exists")
-        found = shutil.which(target) is not None
+        # Build candidate list: for "python" also try "python3" (macOS/Linux)
+        candidates = [target]
+        if target == "python":
+            candidates.append("python3")
+        found = any(shutil.which(c) is not None for c in candidates)
         if found:
             # Concise style: "<name> detected"
             return _create_result("pass", f"{target} detected")

--- a/tests/test_handler_registry.py
+++ b/tests/test_handler_registry.py
@@ -6,6 +6,8 @@ import tempfile
 from typing import Any, cast
 from unittest.mock import patch
 
+from pytest import MonkeyPatch
+
 from azure_functions_doctor.handlers import HandlerRegistry, Rule
 
 
@@ -1163,6 +1165,84 @@ def test_executable_exists_missing_target() -> None:
     assert result["status"] == "fail"
     assert "Missing 'target'" in result["detail"]
 
+
+def test_executable_exists_python3_fallback(monkeypatch: MonkeyPatch) -> None:
+    """Test python target falls back to python3 when python is not on PATH."""
+    registry = HandlerRegistry()
+    rule: Rule = {
+        "id": "test_python3_fallback",
+        "type": "executable_exists",
+        "condition": {
+            "target": "python",
+        },
+    }
+
+    # Simulate: 'python' not found, 'python3' found
+    def fake_which(name: str) -> str | None:
+        return "/usr/bin/python3" if name == "python3" else None
+
+    monkeypatch.setattr("shutil.which", fake_which)
+    result = registry.handle(rule, Path("."))
+    assert result["status"] == "pass"
+    assert "python detected" in result["detail"]
+
+
+def test_executable_exists_python_no_fallback_needed(monkeypatch: MonkeyPatch) -> None:
+    """Test python target found directly — no fallback triggered."""
+    registry = HandlerRegistry()
+    rule: Rule = {
+        "id": "test_python_direct",
+        "type": "executable_exists",
+        "condition": {
+            "target": "python",
+        },
+    }
+
+    def fake_which(name: str) -> str | None:
+        return "/usr/bin/python" if name == "python" else None
+
+    monkeypatch.setattr("shutil.which", fake_which)
+    result = registry.handle(rule, Path("."))
+    assert result["status"] == "pass"
+    assert "python detected" in result["detail"]
+
+
+def test_executable_exists_python_neither_found(monkeypatch: MonkeyPatch) -> None:
+    """Test python target fails when neither python nor python3 on PATH."""
+    registry = HandlerRegistry()
+    rule: Rule = {
+        "id": "test_python_neither",
+        "type": "executable_exists",
+        "condition": {
+            "target": "python",
+        },
+    }
+
+    monkeypatch.setattr("shutil.which", lambda name: None)
+    result = registry.handle(rule, Path("."))
+    assert result["status"] == "fail"
+    assert "not found" in result["detail"]
+
+
+def test_executable_exists_non_python_no_fallback(monkeypatch: MonkeyPatch) -> None:
+    """Test non-python target does NOT trigger python3 fallback."""
+    registry = HandlerRegistry()
+    rule: Rule = {
+        "id": "test_non_python_no_fallback",
+        "type": "executable_exists",
+        "condition": {
+            "target": "func",
+        },
+    }
+
+    # python3 exists but should NOT be tried for 'func'
+    def fake_which(name: str) -> str | None:
+        return "/usr/bin/python3" if name == "python3" else None
+
+    monkeypatch.setattr("shutil.which", fake_which)
+    result = registry.handle(rule, Path("."))
+    assert result["status"] == "fail"
+    assert "not found" in result["detail"]
 
 def test_callable_detection_found() -> None:
     """Test _handle_callable_detection with pattern found."""

